File: 1663.go
func getSmallestString(n int, k int) string {
	ans := make([]byte, n)
	k -= n
	for i := n - 1; i >= 0; i-- {
		t := 0
		for k > 0 && t < 25 {
			k--
			t++
		}
		ans[i] = byte('a' + t)
	}
	return string(ans)
}

File: 1663.java
class Solution {
    public String getSmallestString(int n, int k) {
        char[] ans = new char[n];
        k -= n;
        for(int i = n-1 ;i >= 0; i --) {
            int t = 0;
            while(k > 0 && t < 25) {
                k --;
                t ++;
            }
            ans[i] = (char)('a' + t);
        }
        return new String(ans);
    }
}